docs/reference.cue

```diff
@@ -1 +1,36 @@
 package metadata
```

Comment:
I've been adding comments to this document as if it were specifically written to support the `remap` language/transform, but I now see that it's actually at the top-level `docs/reference.cue`, so I wanted to ask if anything added here is supposed to be used in other parts of Vector as well?